revel/add.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"fmt"
+	"go/build"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"text/template"
+
+	"github.com/revel/revel"
+)
+
+var cmdAdd = &Command{
+	UsageLine: "add [path] [controller_name]",
+	Short:     "add a new controller to the Revel application",
+	Long: `
+Add creates a new controller file and puts it in the Revel application directory you specify and sets up the controller file appropriately. Also adds the views
+directory for the new controller.
+
+For example:
+    revel add import/path/app user
+
+This creates a user.go file in the /app/controllers directory of import/path/app and a User directory under /app/views
+`,
+}
+
+func init() {
+	cmdAdd.Run = addController
+}
+
+var (
+	// revel related paths
+	revelPath  string
+	controller string
+)
+
+func addController(args []string) {
+	// check for args by count. need two arguments.
+	if len(args) == 0 {
+		errorf("The app path and controller name were not supplied. Run 'revel help add' for usage.\n")
+	}
+	if len(args) != 2 {
+		errorf("The incorrect number of arguments was supplied. Run 'revel help add' for usage.\n")
+	}
+
+	controller = args[1]
+
+	// checking and setting go paths
+	initGoPaths()
+
+	// create the path to the application
+	createApplicationPath(args[0])
+
+	// check if the application path exists
+	checkApplicationPath()
+
+	// check if the path to the controllers directory
+	// exists
+	checkControllersDirectory()
+
+	// create the new controller file
+	createControllerFile()
+
+	// create the view directory
+	createControllerViews()
+
+	fmt.Println("Done!")
+}
+
+func createApplicationPath(appName string) {
+	appPath = filepath.Join(srcRoot, filepath.FromSlash(appName))
+}
+
+func checkApplicationPath() {
+	fmt.Printf("Adding controller: %s.go\n", controller)
```

Comment:
I think this is a bit misleading. It claims to be adding a controller but in fact is checking the application path.